src/pathpyG/core/Graph.py

```diff
@@ -256,10 +217,7 @@ def is_edge(self, v: Union[str, int], w: Union[str, int]) -> bool:
             v: source node of edge as integer index or string ID
             w: target node of edge as integer index or string ID       
         """
-        if len(self.node_index_to_id) > 0:
-            return self.node_id_to_index[w] in self._sparse_adj_matrix.getrow(self.node_id_to_index[v]).indices  # type: ignore
-        else:
-            return w in self._sparse_adj_matrix.getrow(v).indices  # type: ignore
+        return self.mapping.to_idx(w) in self._sparse_adj_matrix.getrow(self.mapping.to_idx(v)).indices  # type: ignore
```

Comment:
The same using the PyG edge_index:
```suggestion
        edge_to_check = torch.tensor([self.mapping.to_idx(v), self.mapping.to_idx(w)], dtype=torch.long)
        is_contained = torch.all(torch.eq(edge_index.T, edge_to_check), dim=1)
        return is_contained.any()
```

Reply:
About the inheritance from torch.Data: Indeed, this seems intuitive, and this was what I did in the very first pathpyG implementation. 

However, this brought a lot of downstream complications, as it introduces false semantics about the handling of data object when we start to build our own class hierarchy, i.e. TemporalGraph or HigherOrderGraph inheriting from Graph or the representation of PathData. Moreover, the semantics of some functions in the Data object are rather unclear and partly depend on attributes with special names  that are stored in the object (e.g. num_nodes, attributes with node, edge prefix etc.). I would thus advise against changing it back to my original implementation. Moreover, as we want to support non-tensor attributes, we anyway need additional functionality, e.g. for string or categorical data.

What I agree with is that we should simplify the access to node and edge attributes in the underlying data frame, see also #34. We eventually simply want to support the lookup of node/edge attributes based on IDs that are automatically mapped to indices. There are simpler ways to do this than using the key interface of the classes.

Reply:
And regarding the suggestion for the change to pyG Code. 

The proposed solution will be much slower., as it is not using O(1) operations exclusively. My intention was to be able to get the successors/predecessors without having to iterate through the whole matrix. Hence the use of a scipy sparse matrix, where we can access neighbors based on indices. The solution above requires to check all entries in the matrix, which is prohibitive for large matrices, especially since we have to do it multiple times in a graph traversal algorithm.

Reply:
That is something I did not consider. But we should think about switching to the sparse version of PyTorch: https://pytorch.org/docs/stable/sparse.html#sparse-csr-docs
I think it would be a good idea to add these thoughts about runtime, class design etc. as comments to the code. Then when someone else looks at the code, we do not have to explain them again.